src/main/java/io/github/tryexceptelse/jdex/be/Rolodex.java

```diff
@@ -60,38 +58,51 @@ public void setContacts(ArrayList<Contact> newContacts) {
     }
 
     /**
-     * Returns Contact after adding it to list
-     *
-     * @param newEntries HashMap of ContactEntries, with String keys.
-     *                   This is the map of the entries that will be stored
-     *                   in the new Contact object.
-     * @return Contact which was added to the rolodex. This is so that the UI
-     * has a reference to the newly added object, from which it may display
-     * information to the user, if needed.
+     * @param first:      expects FirstName object representing the first name of the contact
+     * @param last:       expect LastName object representing the last name of the contact
+     * @param email:      expects EmailAddress object representing the email address of the contact
+     * @param streetAddr: expects EmailAddress object represnting the email address of the contact
+     * @param phone:      expects PhoneNumber object representing the phone number of the contact
+     * @param notes:      expects ContactNotes object representing notes for the contact
      */
-    public Contact addContact(HashMap<String, ContactEntry> newEntries) {
-        // placeholder. Contact should be added to contacts.
-        return new Contact(newEntries);
+
+    public Contact addContact(FirstName first,
+                              LastName last,
+                              EmailAddress email,
+                              StreetAddress streetAddr,
+                              PhoneNumber phone,
+                              ContactNotes notes)
+                              throws InvalidObjectException
+    {
+        // Checking to make sure all passed fields are valid. The methods are defined in their own classes
+        for (ContactEntry entry : new ContactEntry[]{first, last, email, streetAddr, phone, notes})
+        {
+            if (!entry.checkHasValidString())
+                throw new InvalidObjectException(entry + "was invalid. Could not add to contacts");
+        }
+        //adding contact to the contatcs ArrayList
+        contacts.add(new Contact(first, last, email, streetAddr, phone, notes));
+        return new Contact();
     }
```

Comment:
addContact, as written, is returning a totally different contact object than the one created using the passed entries.